src/app/metadatafield/metadatafield-list/metadatafield-list.component.ts

```diff
@@ -9,25 +9,67 @@ import { MetadatafieldService } from '../metadatafield.service';
 })
 export class MetadataFieldListComponent implements OnInit {
   public metadataFields: Metadatafield[] = [];
-  public totalmetadataFields = 0;
+  public totalMetadataFields = 0;
+  public pageTotalMetadataFields = 0;
   public errorMessage = '';
+  public hasNext: Boolean;
+  public actualPage: number;
 
-  constructor(private metadatafieldlistcomp: MetadatafieldService) {}
+  constructor(private metadatafieldService: MetadatafieldService) {}
 
   ngOnInit() {
-    this.metadatafieldlistcomp.getAll()
+    if (this.actualPage == null) {
+      this.actualPage = 0;
+    }
+
+    this.getMetadataFieldList();
+    this.getTotalMetadataFields();
+
+  }
+
+  goNext() {
+    if (this.hasNext) {
+      this.actualPage++;
+      console.log(this.actualPage);
+      this.getMetadataFieldList();
+    }
+    console.log(this.hasNext);
+  }
+
+  goPrev() {
+    if (this.actualPage > 0) {
+      this.actualPage--;
+      console.log(this.actualPage);
+      this.getMetadataFieldList();
+    }
+
+  }
+
+  getMetadataFieldList() {
+    this.metadatafieldService.getAll()
```

Comment:
Doing .getAll and then inside .page implies 2 API calls. It is better to do a getAll in ngOnInit the first time the component is loaded. Then, from goPrev goNext call this method that just does .page (no need to do getAll again)